Assets/MRTK/Providers/LeapMotion/LeapMotionArticulatedHand.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using System.Collections.Generic;
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+
+#if LEAPMOTIONCORE_PRESENT
+using Leap.Unity.Attachments;
+using Leap.Unity;
+using Leap;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.LeapMotion.Input
+{
+
+    [MixedRealityController(
+    SupportedControllerType.ArticulatedHand,
+    new[] { Handedness.Left, Handedness.Right })]
+    /// <summary>
+    /// Class that represents one Leap Motion Articulated Hand.
+    /// </summary>
+    public class LeapMotionArticulatedHand : BaseHand
+    {
+        /// <summary>
+        /// Constructor for a Leap Motion Articulated Hand
+        /// </summary>
+        /// <param name="trackingState">Tracking state for the controller</param>
+        /// <param name="controllerHandedness">Handedness of this controller (Left or Right)</param>
+        /// <param name="inputSource">The origin of user input for this controller</param>
+        /// <param name="interactions">The controller interaction map between physical inputs and the logical representation in MRTK</param>
+        public LeapMotionArticulatedHand(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null) 
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+            handDefinition = new ArticulatedHandDefinition(inputSource, controllerHandedness);
+#if LEAPMOTIONCORE_PRESENT
+            SetAttachmentHands();
+#endif
+        }
+
+        private readonly ArticulatedHandDefinition handDefinition;
+
+        // Set the interations for each hand to the Default interactions of the hand definition
+        public override MixedRealityInteractionMapping[] DefaultInteractions => handDefinition?.DefaultInteractions;
+
+#if LEAPMOTIONCORE_PRESENT
+
+        protected LeapMotionDeviceManager LeapMotionDeviceManager => CoreServices.GetInputSystemDataProvider<LeapMotionDeviceManager>();
+
+        protected AttachmentHands AttachmentHands => LeapMotionDeviceManager.attachmentHands;
+
+        // Attachment hand needed for this hand, left or right hand
```

Comment:
I'm a little confused by this comment - "attachment hand needed for this hand, left or right hand" - maybe expand on it a little bit?

Reply:
Yes I can, the leap AttachmentHand contains the joint poses for the current leap hand in frame. There is one AttachmentHand, either left or right, associated with a LeapMotionArticulatedHand.

Reply:
Comment updated